lib/relay/relay_pool.dart

```diff
@@ -316,30 +317,150 @@ class RelayPool {
     }
   }
 
-  /// subscribe shoud be a long time filter search.
-  /// like: subscribe the newest event、notice.
-  /// subscribe info will hold in reply pool and close in reply pool.
-  /// subscribe can be subscribe when new relay put into pool.
-  String subscribe(List<Map<String, dynamic>> filters, Function(Event) onEvent,
-      {String? id}) {
+  /// Subscribes to events matching the given filters across specified relays.
+  ///
+  /// This creates a long-term subscription that will continue receiving events
+  /// until explicitly unsubscribed. This is ideal for monitoring new events or notices.
+  String subscribe(
+    List<Map<String, dynamic>> filters,
+    Function(Event) onEvent, {
+    String? id,
+    List<String>? tempRelays,
+    List<String>? targetRelays,
+    List<int> relayTypes = RelayType.ALL,
+    bool sendAfterAuth = false,
+  }) {
+    // Validate that we have at least one filter
     if (filters.isEmpty) {
       throw ArgumentError("No filters given", "filters");
     }
 
+    // Process and normalize relay addresses
+    if (tempRelays != null) {
+      tempRelays =
+          tempRelays.map((addr) => RelayAddrUtil.handle(addr)).toList();
+    }
+    if (targetRelays != null) {
+      targetRelays =
+          targetRelays.map((addr) => RelayAddrUtil.handle(addr)).toList();
+    }
+
+    // Create subscription and store it in active subscriptions map
     final Subscription subscription = Subscription(filters, onEvent, id);
     _subscriptions[subscription.id] = subscription;
-    send(subscription.toJson());
+
+    // Handle temporary relays first - these are one-time use relays
+    if (tempRelays != null) {
+      for (var tempRelayAddr in tempRelays) {
+        // Try to get existing relay or create a new temporary one
+        Relay? relay = _relays[tempRelayAddr];
+        relay ??= checkAndGenTempRelay(tempRelayAddr);
```

Comment:
Just a suggestion: Since this operator is unique to dart, it might be more readable like this:
```suggestion
        Relay? relay = _relays[tempRelayAddr] ?? checkAndGenTempRelay(tempRelayAddr);
```